File: 2048-build-array-from-permutation/2048-build-array-from-permutation.java
class Solution {
    public int[] buildArray(int[] nums) {
        int ans[]=new int[nums.length];
        HashMap<Integer,Integer> hm=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            hm.put(i,nums[i]);
        }
        for(int j=0;j<ans.length;j++){
            ans[j]=hm.get(hm.get(j));
        }
        return ans;
    }
}

File: 2048-build-array-from-permutation/2048-build-array-from-permutation.cs
public class Solution {
    public int[] BuildArray(int[] nums) {
        int n=nums.Length;
        int[] arr=new int[n];
        for(int i=0;i<nums.Length;i++){
            arr[i]=nums[nums[i]];
        }
        return arr;
    }
}